### ISSUE TITLE
[SOLVED] Tests run and pass, but hang with grey tick/check mark icon (not green)

### ISSUE THREAD

**Issue by BrendanSimon** [kondratyev-nv/vscode-python-test-adapter#277]
I have tests that run and pass but I do not get the green tick icon.
When running I get the blue working icon, but then after the test completes it returns to the grey "untested" icon.

**vscode Environment**

Running vscode on Windows 10 Pro, connecting to Debian 10 Buster Linux VM running in VirtualBox.

```text
Version: 1.64.2 (system setup)
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:28.252Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044
```

**Test**

```python
class Test_Dummy :
    def test_pass( self ) -> None :
        assert True == True
```

**Debug Output:**

```text
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/brendansimon/EFD/brickstrap/efd, configfile: pytest.ini
plugins: cov-3.0.0, mock-3.7.0
collected 1 item
==DISCOVERED TESTS BEGIN==
{"tests": [{"id": "root/opt/sbin/pytest/efd_dummy_test.py::Test_Dummy::test_pass", "line": 50}], "errors": []}
==DISCOVERED TESTS END==

root/opt/sbin/pytest/efd_dummy_test.py .                                 [100%]

============================== 1 passed in 0.04s ===============================
```

**vscode Result**

![vscode-python-test-explorer-issue](https://user-images.githubusercontent.com/472245/154626414-f37bbc9d-56ff-4d5c-a4dd-d22efc57dfcf.PNG)

**More Info (2022-03-02)**

Even though when running in debug mode where the output says it passed, the tests do not *complete*.  The vscode ui states the tests are still running ("Running tests...") - even a simple test that just asserts true.

So looks like something is waiting indefinitely with Python Test Explorer.

![vscode-python-test-explorer-running-tests](https://user-images.githubusercontent.com/472245/156284283-980bf7e7-080d-42f5-b373-7a8556c8d02f.png)

![vscode-python-test-explorer-running-tests-2](https://user-images.githubusercontent.com/472245/156284328-b1680cd5-5dc1-482e-9399-d3889368e9c7.png)

**NOTE:** running with the same tests with the vscode native python test framework is ok.  I do get green ticks (check marks).

![vscode-python-test-passed-1](https://user-images.githubusercontent.com/472245/156284925-1a13b954-73ae-436c-9115-5e86005e31b5.png)

![vscode-python-test-passed-2](https://user-images.githubusercontent.com/472245/156284937-1b8ab45a-b86a-468b-978f-c0bca76e05f5.png)

**Comment by BrendanSimon** [comment]
So I switched branches and things started working with Python Test Explorer.  All tests pass now except for 1 (with the same symptoms of greyed out result - not red, not greed).

I have switched back and forth between the prior branch and made sure the vscode settings were the same, and I now get the same partial test results (but everything passes in the native python test explorer).

![vscode-python-test-explorer-issue-2](https://user-images.githubusercontent.com/472245/157800942-c3b0cafa-0c57-454a-bc35-3a95b60514d4.png)

VS Code is now at the following version.

```
Version: 1.65.1 (system setup)
Commit: 8908a9ca0f221f36507231afb39d2d8d1e182702
Date: 2022-03-08T02:06:27.846Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044

============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-7.0.1, pluggy-1.0.0 -- /home/brendansimon/EFD/brickstrap/ind-efd/venv_3/bin/python
cachedir: .pytest_cache
rootdir: /home/brendansimon/EFD/brickstrap/ind-efd, configfile: pytest.ini
plugins: cov-3.0.0, mock-3.7.0, pyfakefs-4.5.5
collecting ... collected 1 item
==DISCOVERED TESTS BEGIN==
...
```

**Comment by BrendanSimon** [comment]
**SOLVED !!**

I noticed that the debug output had references to files stored in `/mnt/brendansimon/...`

My home directory was on a separate (virtual) disk, mouted to `/mnt/brendansimon` and then symlinked from `/home/brendansimon`.

The symlink was causing an issue with my build so I removed that mounted directly to `/home/brendansimon` via `/etc/fstab`, and the build works.

Seems there are some references to `../../../../../mnt/brendansimon/....../pytest/efd_weather_test.py` in the pytest cache which seems to cause havoc with Python Test Explorer (in run mode only - debug seems ok).

Anyway, removing the cahce directory fixed things so that Python Test Explorer now works as expected.

Sorry for all the noise :-/
